### PR TITLE
Pass args and kwargs around so that the get_file() method can respond to URL parameters.

### LINKED ISSUE #120: Pass args & kwargs around so that get_file() can implement URL-based logic.
Currently you can't access URL args & kwargs from the get_file() method forcing more of the class to be overridden than is ideal.

### PR DESCRIPTION
**Pull request by willson556**
Resolves #120.

### CODE REVIEW

**Comment by benoitbryon** [comment]
Hi @willson556,

> the get_file() method can respond to URL parameters

What about using view's ``self.args`` and ``self.kwargs`` in ``get_file()`` method?

View's attributes such as ``request``, ``args``, and ``kwargs`` are set by Django's ``as_view()`` magic: see https://github.com/django/django/blob/1.9.6/django/views/generic/base.py#L66-L67

As an example, if URLConf looks like:

```python
year_view = sampleviews.YearDownloadView.as_view()

urlpatterns = [
    url(r'^files/(?P<year>[0-9]{4})/$', views.year_archive),
]
```

... then you should be able to use "year" from ``self.kwargs``:

```python
class YearDownloadView(VirtualDownloadView):
    def get_file(self):
        year = self.kwargs['year']
        # ... Do something with year...
```

Does it apply to your use case? Else can you tell me more about the situation where ``*args`` and ``**kwargs`` are mandatory?

File: django_downloadview/response.py [hunk 1]
```diff
@@ -117,9 +117,10 @@ class DownloadResponse(StreamingHttpResponse):
       attributes (size, name, ...).
 
     """
+
     def __init__(self, file_instance, attachment=True, basename=None,
                  status=200, content_type=None, file_mimetype=None,
-                 file_encoding=None):
+                 file_encoding=None, extra_headers=None, **kwargs):
```
Comment (benoitbryon) [inline django_downloadview/response.py]: Does ``**kwargs`` come from https://github.com/benoitbryon/django-downloadview/pull/109 ?
Is it really related to the use of ``args`` and ``kwargs`` in ``get_file()``?

About ``extra_headers``, I don't see the relationship with ``args`` and ``kwargs`` in ``get_file()``. If it is another feature, can you create another pull-request for this particular change?

**Comment by willson556** [comment]
I had missed that Django saves the args and kwargs as instance variables. That does resolve my specific use case. 

With that knowledge, my only argument for including this PR would be consistency with other CBV's in Django which almost all pass the args and kwargs around (for better or worse) to almost all commonly overriden methods.

File: django_downloadview/response.py (hunk 1 above)
Comment (willson556) [inline django_downloadview/response.py]: You're right, those are unrelated. `**kwargs` did come in from #109 and isn't necessary. And `extra_headers` came from some preliminary work on #122 that I didn't intend to include. I'll remove those unrelated changes.

**Pull request closed without merging**